app/src/pages/p/[projectSlug]/usage/index.tsx

```diff
@@ -16,72 +16,240 @@ import {
+
+  const stats = useStats(selectedProject?.id || "", startDate, endDate);
+
+  const { totalInferenceSpend, totalTrainingSpend, totalInputTokens, totalOutputTokens } =
+    useMemo(() => {
+      const calculateTotal = <T extends Record<string, any>>(
+        dataArray: T[] | undefined,
+        key: keyof T,
+      ) => dataArray?.reduce((acc: number, cur: T) => acc + Number(cur[key]), 0) ?? 0;
+
+      return {
+        totalTrainingSpend: calculateTotal(stats.data?.periods, "trainingCost"),
+        totalInferenceSpend: calculateTotal(stats.data?.periods, "inferenceCost"),
+        totalInputTokens: calculateTotal(stats.data?.fineTunes, "inputTokens"),
+        totalOutputTokens: calculateTotal(stats.data?.fineTunes, "outputTokens"),
+      };
+    }, [stats.data]);
+
+  const updateMonth = (operation: "add" | "subtract") => {
+    setStartDate(dayjs(startDate)[operation](1, "month").startOf("month").format("YYYY-MM-DD"));
+    setEndDate(dayjs(endDate)[operation](1, "month").endOf("month").format("YYYY-MM-DD"));
+  };
+
+  const nextMonthIsClickable = () => dayjs(startDate).add(1, "month").isBefore(dayjs());
+
+  const handleNextMonthChange = () => {
+    if (!nextMonthIsClickable()) return;
+    updateMonth("add");
+  };
+
+  const handlePrevMonthChange = () => {
+    updateMonth("subtract");
+  };
 
   return (
     <AppShell title="Usage" requireAuth>
-      <VStack px={8} py={8} alignItems="flex-start" spacing={4}>
-        <Text fontSize="2xl" fontWeight="bold">
-          Usage
-        </Text>
-        <VStack margin="auto" spacing={4} align="stretch" w="full">
-          <HStack gap={4} align="start">
-            <Card flex={1}>
-              <CardBody>
-                <UsageGraph />
-              </CardBody>
+      <VStack px={8} py={8} alignItems="flex-start">
+        <Tabs w="full">
+          <HStack justifyContent="space-between" marginX={0}>
+            <TabList>
+              <Tab>
+                <Heading size="md">Cost</Heading>
+              </Tab>
+              <Tab>
+                <Heading size="md">Activity</Heading>
+              </Tab>
+            </TabList>
+            <Spacer />
+            <Card w={"180px"} colorScheme="white" color={"gray.900"} padding={"5px"}>
+              <HStack spacing={0} justifyContent="space-between">
+                <Icon
+                  onClick={handlePrevMonthChange}
+                  cursor={"pointer"}
+                  as={ChevronLeftIcon}
+                  boxSize={5}
+                  color={"gray.900"}
+                />
+                <Text onClick={handlePrevMonthChange} cursor={"pointer"}>
+                  {formatToUTCDayMonth(startDate || startOfThisMonth)}
+                </Text>
+                <Text paddingX={"5px"}> - </Text>
+                <Text onClick={handleNextMonthChange} cursor={"pointer"}>
+                  {formatToUTCDayMonth(endDate || endOfThisMonth)}
+                </Text>
+                <Icon
```

Comment:
Would it be possible to use a predefined interactive component like [IconButton](https://chakra-ui.com/docs/components/icon-button/usage) with the `disabled` prop for the behavior we want? Defining our own input behavior like this is a bit fraught, because it's easy to forget about minor UX niceties that are noticeable to the end user -- eg. in this case the cursor remains as pointer even if the button is disabled, which sends the user a contradictory signal.

Reply:
IconButton should be good, changing it!